yb-voyager/src/pgss/parser.go

```diff
@@ -150,3 +163,108 @@ func parseFloatOrZero(value, fieldName string, target *float64) (err error) {
 	}
 	return
 }
+
+// parseJSONBRecord parses a JSONB string and extracts PgStatStatements fields
+func parseJSONBRecord(jsonStr string) (*PgStatStatements, error) {
+	var jsonData map[string]interface{}
+	err := json.Unmarshal([]byte(jsonStr), &jsonData)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse JSONB: %w", err)
+	}
+
+	entry := &PgStatStatements{}
+
+	// Extract required fields
+	entry.QueryID, err = getInt64FromJSON(jsonData, "queryid")
+	if err != nil {
+		return nil, err
+	}
+
+	entry.Query, err = getStringFromJSON(jsonData, "query")
+	if err != nil {
+		return nil, err
+	}
+
+	entry.Calls, err = getInt64FromJSON(jsonData, "calls")
+	if err != nil {
+		return nil, err
+	}
+	if entry.Calls <= 0 {
+		return nil, fmt.Errorf("invalid calls: %d", entry.Calls)
+	}
+
+	// Optional field - rows
+	entry.Rows, _ = getInt64FromJSON(jsonData, "rows")
+
+	// Handle timing columns - check both new naming (total_exec_time) and old naming (total_time)
+	entry.TotalExecTime = getFloat64OrZero(jsonData, "total_exec_time", "total_time")
+	entry.MeanExecTime = getFloat64OrZero(jsonData, "mean_exec_time", "mean_time")
+	entry.MinExecTime = getFloat64OrZero(jsonData, "min_exec_time", "min_time")
+	entry.MaxExecTime = getFloat64OrZero(jsonData, "max_exec_time", "max_time")
+	entry.StddevExecTime = getFloat64OrZero(jsonData, "stddev_exec_time", "stddev_time")
+
+	return entry, nil
+}
+
+// getInt64FromJSON extracts an int64 value from JSON data
+func getInt64FromJSON(data map[string]interface{}, key string) (int64, error) {
+	val, ok := data[key]
+	if !ok {
+		return 0, fmt.Errorf("missing required field: %s", key)
+	}
+
+	switch v := val.(type) {
+	case float64:
+		return int64(v), nil
+	case int64:
+		return v, nil
+	case int:
+		return int64(v), nil
+	case string:
+		return strconv.ParseInt(v, 10, 64)
+	default:
+		return 0, fmt.Errorf("invalid type for %s: %T", key, val)
+	}
+}
+
+// getStringFromJSON extracts a string value from JSON data
+func getStringFromJSON(data map[string]interface{}, key string) (string, error) {
+	val, ok := data[key]
+	if !ok {
+		return "", fmt.Errorf("missing required field: %s", key)
+	}
+
+	str, ok := val.(string)
+	if !ok {
+		return "", fmt.Errorf("invalid type for %s: expected string, got %T", key, val)
+	}
+
+	str = strings.TrimSpace(str)
+	if str == "" {
+		return "", fmt.Errorf("empty value for %s", key)
+	}
+
+	return str, nil
+}
+
+// getFloat64OrZero extracts a float64 value from JSON data, trying multiple keys
+// Returns 0.0 if none of the keys exist (for optional fields)
+func getFloat64OrZero(data map[string]interface{}, keys ...string) float64 {
+	for _, key := range keys {
+		if val, ok := data[key]; ok {
+			switch v := val.(type) {
+			case float64:
+				return v
+			case int64:
+				return float64(v)
+			case int:
+				return float64(v)
+			case string:
+				if f, err := strconv.ParseFloat(v, 64); err == nil {
+					return f
+				}
+			}
+		}
+	}
+	return 0.0
+}
```

Comment:
These three `get<type>FromJSON` functions can be utility functions

Reply:
Done